### PR TITLE
update: set User-Agent header in GCS storage driver

### PR DESCRIPTION
**Pull request by milosgajdos**
We also clean up some redundant code.

### CODE REVIEW

File: registry/storage/driver/gcs/gcs.go [hunk 1]
```diff
@@ -165,10 +166,7 @@ func FromParameters(ctx context.Context, parameters map[string]interface{}) (sto
 			return nil, err
 		}
 		ts = jwtConf.TokenSource(ctx)
-		gcs, err = storage.NewClient(ctx, option.WithCredentialsFile(fmt.Sprint(keyfile)))
-		if err != nil {
-			return nil, err
-		}
+		options = append(options, option.WithCredentialsFile(fmt.Sprint(keyfile)))
```
Comment (thaJeztah) [inline registry/storage/driver/gcs/gcs.go:169]: The `fmt.Sprint()` everywhere make my eyes bleed 😂

**Review by thaJeztah — APPROVED** [review]
LGTM

File: registry/storage/driver/gcs/gcs.go (hunk 1 above)
Comment (milosgajdos) [reply registry/storage/driver/gcs/gcs.go:169]: The whole config file needs rework in v4. I'm pretty sure I saw an issue opened by @stevvooe last century. It's somewhere in the backlog

Comment (thaJeztah) [reply registry/storage/driver/gcs/gcs.go:169]: Yeah, I spotted it in your changed lines, but then saw it was all over the place, so don't bother for this PR

**Review by deleteriousEffect — APPROVED** [review]

**Comment by deleteriousEffect** [comment]
Good stuff :rocket:

**Pull request merged**